ovara-backend/src/main/scala/fi/oph/ovara/backend/raportointi/Controller.scala

```diff
@@ -1,35 +1,115 @@
+) {
+  val LOG: Logger = LoggerFactory.getLogger(classOf[Controller]);
 
   @Value("${ovara.ui.url}")
   val ovaraUiUrl: String = null
-  
+
+  private val mapper = new ObjectMapper()
+  mapper.registerModule(DefaultScalaModule)
+  mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+  mapper.configure(SerializationFeature.INDENT_OUTPUT, true)
 
   @GetMapping(path = Array("ping"))
   def ping = "Ovara application is running!"
 
   @GetMapping(path = Array("user"))
-  def user(@AuthenticationPrincipal userDetails: UserDetails): UserResponse = {
-    UserResponse(
-      user = if (userDetails == null) null else UserResponseUser(userOid = userDetails.getUsername)
+  def user(@AuthenticationPrincipal userDetails: UserDetails): String = {
+    mapper.writeValueAsString(
+      UserResponse(
+        user =
+          if (userDetails == null)
+            null
+          else
+            val asiointikieli = onrService.getAsiointikieli(userDetails.getUsername) match
+              case Left(e) => None
+              case Right(v) => Some(v)
+
+            User(
+              userOid = userDetails.getUsername,
+              authorities = AuthoritiesUtil.getRaportointiAuthorities(userDetails.getAuthorities),
+              asiointikieli = asiointikieli
+            )
+      )
     )
   }
 
   @GetMapping(path = Array("login"))
   def login = RedirectView(ovaraUiUrl)
+
+  @GetMapping(path = Array("csrf"))
+  def csrf(csrfToken: CsrfToken): String = mapper.writeValueAsString(csrfToken)
+
+  @GetMapping(path = Array("alkamisvuodet"))
+  def alkamisvuodet: String = mapper.writeValueAsString(commonService.getAlkamisvuodet)
+
+  @GetMapping(path = Array("haut"))
+  def haut: String = mapper.writeValueAsString(commonService.getHaut)
+
+  @GetMapping(path = Array("koulutukset-toteutukset-hakukohteet"))
+  def koulutukset_toteutukset_hakukohteet(
+      @RequestParam("alkamiskausi") alkamiskausi: java.util.Collection[String],
+      @RequestParam("haku") haku: java.util.Collection[String],
+      @RequestParam("koulutuksenTila", required = false) koulutuksenTila: String,
+      @RequestParam("toteutuksenTila", required = false) toteutuksenTila: String,
+      @RequestParam("hakukohteenTila", required = false) hakukohteenTila: String,
+      @RequestParam("valintakoe", required = false) valintakoe: Boolean,
+      response: HttpServletResponse
+  ): Unit = {
+    val maybeKoulutuksenTila = Option(koulutuksenTila)
+    val maybeToteutuksenTila = Option(toteutuksenTila)
+    val maybeHakukohteenTila = Option(hakukohteenTila)
+    val maybeValintakoe      = Option(valintakoe)
+    val wb = koulutuksetToteutuksetHakukohteetService.
+      get(
+      alkamiskausi.asScala.toList,
+      haku.asScala.toList,
+      maybeKoulutuksenTila,
+      maybeToteutuksenTila,
+      maybeHakukohteenTila,
+      maybeValintakoe
+    )
+    try {
+      LOG.info(s"Sending excel in the response")
```

Comment:
Näitä varmaan voi sitten myöhemmin täsmentää että minkä apin excel kyseessä kun tulee useampia